OpenRA.Mods.Common/Traits/Player/PlayerStatistics.cs

```diff
@@ -199,14 +207,19 @@ void INotifyKilled.Killed(Actor self, AttackInfo e)
 			{
 				defenderStats.ArmyValue -= cost;
 				includedInArmyValue = false;
+				playerStats.Units[self.Info.Name].Count--;
```

Comment:
We can fix the missing icon crash (or duplication, if we found another way to source the icon) by defining a `public string OverrideActor = null;` in `UpdatesPlayerStatisticsInfo` and then keying this and the other places on `info.OverrideActor ?? self.Info.Name`.

Actors (or override actors) that don't have the `Buildable` trait should never be added to the cache.